components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -29,14 +29,14 @@ const MAX_MINUTES: u32 = 59;
 const MAX_SECONDS: u32 = 59;
 const MAX_MICROS: u32 = 999_999;
 
+/// `MAX_TIME_IN_SECS` is the maximum for mysql time type.
+const MAX_TIME_IN_SECS: i64 =
+    (MAX_HOURS * SECS_PER_HOUR + MAX_MINUTES * SECS_PER_MINUTE + MAX_SECONDS) as i64;
+
 #[inline]
 fn check_hour(hour: u32) -> Result<u32> {
     if hour > MAX_HOURS {
-        Err(invalid_type!(
-            "invalid hour value: {} larger than {}",
-            hour,
-            MAX_HOURS
-        ))
+        Err(Error::overflow(hour, MAX_HOURS))
```

Comment:
I find the `Error::overflow` code is:
```
pub fn overflow(data: impl Display, expr: impl Display) -> Error {
    let msg = format!("{} value is out of range in '{}'", data, expr);
    Error::Eval(msg, ERR_DATA_OUT_OF_RANGE)
}
```

It doesn't seem correct to me that `MAX_HOURS` is what the `expr` is expected to be here.